libr/util/str.c

```diff
@@ -3564,6 +3565,26 @@ R_API const char *r_str_str_xy(const char *s, const char *word, const char *prev
 	return d;
 }
 
+R_API size_t r_charset_encode_str(ut8 *asciistr, ut8 *in, size_t len_input, RCharset *r_char, size_t len_charset) {
+	bool is_in_set;
+	size_t i, j;
+	for (i = 0; i < len_input; i ++) {
+		is_in_set = false;
+		for (j = 0; j < len_charset; j ++) {
```

Comment:
just use the sdb hashtable directly, it will be a single lookup instead of iterating over all the charsets, or create a wrapper function that does that. Right now your apporach is just a byte translation. Some encodings will be multibyte, this is, reading the string evcery 2 bytes and using the endianness settings to find the given multibyte result. So the charset can have an array of word sizes, and the value could be more than one byte. How does it sound? you can try decoding first one byte as key from the sdb and then check for 2 bytes if the format specifies that (this can be another key in the database)

Reply:
I do not understand what is expected. I think I have to use `sdb_ht_find`. The first argument is an hash table.  I imagine something like that:

`HtPP *hash_table = NULL;
hash_table = sdb_ht_new ();
sdb_ht_find (hash_table, 0x7f, NULL);`

I do not see the benefit to create a new hashtable and feed it in the same function to finally find an element in.

Or I might replace RCharset by hash_table.


Can you clarify please?

Reply:
![ksnip_20200809-163935](https://user-images.githubusercontent.com/5904302/89734871-12c6fb00-da5f-11ea-9c14-d86d8ea5fdca.png)

@trufae maybe a mix of both, i can try to realize that on top of sdb